### PR TITLE
Publish Checkstyle Results

### PR DESCRIPTION
**Pull request by magicgoody**
This PR was created to add an action that visualize the results of Checkstyle tests and adds them to the workflow summary page.

The action requires additional permissions:
checks: write (allow to add test result to workflow summary page)

Example runs:
no error: https://github.com/magicgoody/beam/actions/runs/5964470451
with error: https://github.com/magicgoody/beam/actions/runs/5949179444

Example screenshots:
Checkstyle result without violations
image

Checkstyle result with violations
image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Review by aleksandr-dudko — APPROVED** [review]

File: .github/workflows/beam_PreCommit_Spotless.yml [hunk 1]
```diff
@@ -94,4 +94,9 @@ jobs:
         uses: actions/upload-artifact@v3
         with:
           name: java-code-coverage-report
-          path: "**/build/reports/checkstyle/*.xml"
\ No newline at end of file
+          path: "**/build/reports/checkstyle/*.xml"
+      - name: Publish checkstyle check
+        uses: jwgmeligmeyling/checkstyle-github-action@master
```
Comment (damccorm) [inline .github/workflows/beam_PreCommit_Spotless.yml:99]:
```suggestion
        uses: jwgmeligmeyling/checkstyle-github-action@v1
```

Generally when we use 3rd party actions, we should pin to a tag. This is less likely to randomly break, and the infra team will only allowlist actions that are pinned to tags

Comment (damccorm) [reply .github/workflows/beam_PreCommit_Spotless.yml:99]: I created a jira ticket with infra (may not be publicly visible) to allow list this action @ v1 https://issues.apache.org/jira/browse/INFRA-24948

**Review by damccorm — COMMENTED** [review]
Thanks!

File: .github/workflows/beam_PreCommit_Spotless.yml (hunk 1 above)
Comment (magicgoody) [reply .github/workflows/beam_PreCommit_Spotless.yml:99]: Okay, thank you!

Comment (damccorm) [reply .github/workflows/beam_PreCommit_Spotless.yml]: The ticket was approved, merging

**Pull request merged**